cms/models/pluginmodel.py

```diff
@@ -48,6 +48,17 @@ def __new__(cls, name, bases, attrs):
         # create a new class (using the super-metaclass)
         new_class = super(PluginModelBase, cls).__new__(cls, name, bases, attrs)
 
+        # Set the one-to-one parent-child field to not create the reverse
+        # related descriptor to avoid name clashes for plugins with the same
+        # fieldnames. Skip this check if the parent-child field has been explicitly
+        # defined in the plugin subclass
+        if 'cmsplugin_ptr' not in attrs:
+            for field  in new_class._meta.fields:
+                if field.name == 'cmsplugin_ptr':
+                    field.rel.related_name = '{0}_{1}'.format(
```

Comment:
wouldn't it be better to leverage django's supported syntax https://docs.djangoproject.com/en/1.7/topics/db/models/#be-careful-with-related-name.

I wonder if it would work at this stage though..